azurerm/internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -1152,3 +1212,75 @@ func resourceAzureRMCosmosDBAccountVirtualNetworkRuleHash(v interface{}) int {
 
 	return schema.HashString(buf.String())
 }
+
+func expandCosmosdbAccountBackupPolicy(input []interface{}) (documentdb.BasicBackupPolicy, error) {
+	if len(input) == 0 || input[0] == nil {
+		return documentdb.BackupPolicy{
+			Type: documentdb.TypeBackupPolicy,
+		}, nil
+	}
+	attr := input[0].(map[string]interface{})
+
+	switch attr["type"].(string) {
+	case string(documentdb.TypeContinuous):
+		if v := attr["interval_in_minutes"].(int); v != 0 {
+			return nil, fmt.Errorf("`interval_in_minutes` can not be set when `type` in`backup_policy` is `Continuous` ")
+		}
+		if v := attr["retention_in_hours"].(int); v != 0 {
+			return nil, fmt.Errorf("`retention_in_hours` can not be set when `type` in`backup_policy` is `Continuous` ")
+		}
+		return documentdb.ContinuousModeBackupPolicy{
+			Type: documentdb.TypeContinuous,
+		}, nil
+
+	case string(documentdb.TypePeriodic):
+		return documentdb.PeriodicModeBackupPolicy{
+			Type: documentdb.TypePeriodic,
+			PeriodicModeProperties: &documentdb.PeriodicModeProperties{
+				BackupIntervalInMinutes:        utils.Int32(int32(attr["interval_in_minutes"].(int))),
```

Comment:
Do we need to test whether the `interval_in_minutes` is present in the config here?

Reply:
0 is not a valid value for the interval in minutes. The service accepts 0 and returns a default value for the interval, I'll add a validation in the test if there is a default value for the interval.